services/streammanager/kafka/kafkamanager.go

```diff
@@ -248,6 +248,24 @@ func NewProducer(destConfigJSON interface{}, o Opts) (*producerImpl, error) { //
 		return nil, fmt.Errorf("invalid configuration: %w", err)
 	}
 
+	convertToAvro := destConfig.ConvertToAvro
+	avroSchemas := destConfig.AvroSchemas
+	var codecs map[string]goavro.Codec
+	if convertToAvro {
+		codecs = make(map[string]goavro.Codec, len(avroSchemas))
+		for i, avroSchema := range avroSchemas {
+			if len(avroSchema.SchemaId) > 0 {
```

Comment:
Optional semplification, you could remove the `if/else` and do it with a single `if`:

```go
if len(avroSchema.SchemaId) == 0 {
    return nil, fmt.Errorf("unable to create codec for schemaId:%+v, with error: %w", avroSchema.SchemaId, err)
}

newCodec, err := goavro.NewCodec(avroSchema.Schema)
// ...
```

Reply:
Here the two error messages are different though I can omit else.

Reply:
Here two error messages are different. But I have removed the `else`.